tool/tsh/common/tsh.go

```diff
@@ -3602,6 +3620,28 @@ func onSSH(cf *CLIConf) error {
 		return trace.Wrap(err)
 	}
 
+	// If "tsh ssh -V" is invoked, tsh is in OpenSSH compatibility mode, show
+	// the version and exit.
+	if cf.ShowVersion {
+		modules.GetModules().PrintVersion()
+		return nil
+	}
+
+	// If "tsh ssh" is invoked with the "-t" or "-T" flag, manually validate
+	// "-t" and "-T" flags for "tsh ssh" due to the lack of inverse short flags
+	// in kingpin.
+	if cf.Interactive && cf.NonInteractive {
+		return trace.BadParameter("either -t or -T can be specified, not both")
+	}
+	if cf.NonInteractive {
+		cf.Interactive = false
+	}
+
+	// If "tsh ssh" is invoked the user must specify some host to connect to.
+	if cf.UserHost == "" {
```

Comment:
I don't follow why we had to make the argument optional in kingpin and validate it here. Can we leave a comment explaining why this was necessary?

Reply:
It's to support `tsh ssh -V` which can run without any additional arguments. I'll update the comment to explain why I had to remove `Required()` from that `Arg`.

Reply:
Does this affect the usage docs badly? I reckon that everyone would likely figure out that sshing into an unspecified destination is never going to work, but still...

Reply:
It does.

Adding `Required()` removes the square brackets from `--help` output.

Not ideal, but I can update the help text to indicate this is a required argument.

Output of `tsh ssh --help` before:

```
$ tsh ssh --help
usage: tsh ssh [<flags>] <[user@]host> [<command>...]

Run shell or execute a command on a remote SSH node.

Flags:
[...]

Args:
  <[user@]host>  Remote hostname and the login to use
  [<command>]    Command to execute on a remote host

Aliases:
```

Output of `tsh ssh --help` after this PR:

```
$ tsh ssh --help
usage: tsh ssh [<flags>] [<[user@]host>] [<command>...]

Run shell or execute a command on a remote SSH node.

Flags:
[...]

Args:
  [<[user@]host>]  Remote hostname and the login to use, this argument is required.
  [<command>]      Command to execute on a remote host

Aliases:
```